src/java/org/apache/cassandra/config/EncryptionOptions.java

```diff
@@ -182,8 +182,9 @@ static Set<String> asSet()
         {
             Set<String> valueSet = new HashSet<>();
             ConfigKey[] values = values();
-            for(ConfigKey key: values) {
-                valueSet.add(key.getKeyName().toLowerCase());
+            for (ConfigKey key : values)
+            {
+                valueSet.add(key.toString().toLowerCase());
```

Comment:
@maulin-vasavada use `org.apache.cassandra.utils.LocalizeString` for `toLowerCase`. This will not compile on `trunk` anymore.

Reply:
For this I've to rebase it seems. I don't see that in my local.

Reply:
Can we leave this until all other comments are resolved from you and other reviewers? Based on my past experience, I would take the latest trunk and reapply all the changes vs rebasing to keep commit history clear on this PR.

Reply:
@maulin-vasavada could you please resolve this? Then I will build this branch in CircleCI to see where we are with it as a whole.

Reply:
Yes, will get to this soon.

Reply:
@smiklosovic How do you suggest I do this? I was thinking of doing following,

1. get the latest trunk to my local (already done)
2. create another branch - let us say - `CASSANDRA-18508-sensitive-jmx-ssl-config-2` from the latest trunk
3. cherry-pick the squashed commit from the branch `CASSANDRA-18508-sensitive-jmx-ssl-config` to the above branch and change this PR to be from the `CASSANDRA-18508-sensitive-jmx-ssl-config-2`

Let me know if there are better suggestions.

Reply:
I managed to find a way out-

1. I got the latest trunk locally
2. I created a temp branch out of it
3. cherry-picked commit(s) from the `CASSANDRA-18508-sensitive-jmx-ssl-config` branch to the temp branch
4. resolved any conflicts in the `EncryptionOptions.java`
5. force renamed the temp branch `git branch -f -m  CASSANDRA-18508-sensitive-jmx-ssl-config`
6. fetched from myfork/`CASSANDRA-18508-sensitive-jmx-ssl-config` branch and resolved the same conflict
7. pushed the local branch to myfork to update this PR

This seem to have worked. Please review and we can move to Doug for more reviews.

Reply:
git fetch upstream
git checkout trunk
git merge upstream/trunk
git checkout _this_branch_
git rebase trunk
((( resolve conflicts)))
git push origin _this_branch_ -f

Reply:
@maulin-vasavada 

Your branch contains merge commits etc, that will not be deliverable on the merge. It has to be just one commit. I put all the changes into this branch.

https://github.com/instaclustr/cassandra/commit/1d172ce58f2587ab521696e6037ae8ab6bc16c94

you can hard reset all your changes in your branch and cherry-pick that one commit and force-push it to  your remote to be on par with it.

Please create one commit for this branch so everybody has easier life.

Reply:
@maulin-vasavada 

https://app.circleci.com/pipelines/github/instaclustr/cassandra/4958/workflows/134abde9-5c0d-47f6-82d5-d02f9be48078


Fails on 

`com.datastax.driver.core.exceptions.ServerError: An unexpected error occurred server side on localhost/127.0.0.1:46255: org.yaml.snakeyaml.error.YAMLException: Failed calling getter for property 'optional' on class org.apache.cassandra.config.EncryptionOptions`

```
  ✕ j17_unit_tests                                  15m 17s
      org.apache.cassandra.db.virtual.SettingsTableTest testTransparentEncryptionOptionsOverride
      org.apache.cassandra.db.virtual.SettingsTableTest testArray
      org.apache.cassandra.db.virtual.SettingsTableTest testSelectAll
      org.apache.cassandra.db.virtual.SettingsTableTest testSelectEmpty
      org.apache.cassandra.db.virtual.SettingsTableTest testAuditOverride
      org.apache.cassandra.db.virtual.SettingsTableTest testSelectOverride
      org.apache.cassandra.db.virtual.SettingsTableTest testSelectPartition
      org.apache.cassandra.db.virtual.SettingsTableTest testEncryptionOverride
      org.apache.cassandra.config.PropertiesTest backAndForth
```

Reply:
@smiklosovic I fixed that test.

Reply:
Once everything looks I'll squash commits.